VMEncryption/main/check_util.py

```diff
@@ -185,11 +185,22 @@ def validate_lvm_os(self, public_settings):
         if detected:
             raise Exception("LVM OS disk layout does not satisfy prerequisites ( see https://aka.ms/adelvm )")
 
-    def precheck_for_fatal_failures(self, public_settings):
+    def validate_memory_os_encryption(self, public_settings, encryption_status):
+        is_enable_operation = False
+        encryption_operation = public_settings.get(CommonVariables.EncryptionEncryptionOperationKey)
+        if encryption_operation in [CommonVariables.EnableEncryption, CommonVariables.EnableEncryptionFormat, CommonVariables.EnableEncryptionFormatAll]:
+            is_enable_operation = True
+        volume_type = public_settings.get(CommonVariables.VolumeTypeKey)
+        if is_enable_operation and not volume_type.lower() == CommonVariables.VolumeTypeData.lower() and encryption_status["os"] == "NotEncrypted":
```

Comment:
on first encrypt, is encryption_status always non-null and populated with "NotEncrypted" string?  May be good to replace this with a constant to keep in synch with other locations in code that might set it, rather than hardcoded string "NotEncrypted"

Reply:
By default the encryption status is set to NotEncrypted.
Ref: https://github.com/Azure/azure-linux-extensions/blob/master/VMEncryption/main/DiskUtil.py#L546